### PR TITLE
rough in switchEthereumChain

### PR DESCRIPTION
**Pull request by brad-decker**
Implements basic swithEthereumChain feature for suggesting a user to switch to a specified chain that **they already have added to MetaMask**. This method will work with default networks like mainnet.

### CODE REVIEW

File: app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+    try {
+      await updateRpcTarget(
+        await requestUserApproval({
+          origin,
+          type: MESSAGE_TYPE.SWITCH_ETHEREUM_CHAIN,
+          requestData: {
+            rpcUrl: existingNetwork.rpcUrl,
+            chainId: existingNetwork.chainId,
+            nickname: existingNetwork.nickname,
```
Comment (shanejonas) [inline app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js:80]: [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) has `chainName`

Comment (brad-decker) [reply app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js:80]: true, but internally we call it nickname. existingNetwork comes from our state. EIP-3085 is implemented in the add-ethereum-chain file that gets the chainName from the dapp.

File: app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            ticker: existingNetwork.ticker,
```
Comment (shanejonas) [inline app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js:81]: EIP-3085 uses:

```
  nativeCurrency?: {
    name: string;
    symbol: string;
    decimals: number;
  };
  ```
  
  not sure if this is the same, or maybe just `symbol` and it has to match

Comment (shanejonas) [reply app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js:81]: ah, only `chainId` is from the dapp

File: app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    );
+  }
+
+  if (!isSafeChainId(parseInt(_chainId, 16))) {
+    return end(
+      ethErrors.rpc.invalidParams({
+        message: `Invalid chain ID "${_chainId}": numerical value greater than max safe value. Received:\n${chainId}`,
+      }),
+    );
+  }
```
Comment (rekmarks) [inline app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js:62]: I don't think this check is necessary for the purposes of this method, but it's nice from a "fail early and often" perspective.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**